commons/src/main/java/io/aiven/kafka/connect/common/source/input/ByteArrayTransformer.java

```diff
@@ -18,70 +18,65 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.Arrays;
 import java.util.Map;
-import java.util.Spliterator;
-import java.util.Spliterators;
-import java.util.stream.Stream;
-import java.util.stream.StreamSupport;
+import java.util.function.Consumer;
 
 import org.apache.kafka.common.config.AbstractConfig;
 
+import org.apache.commons.io.IOUtils;
 import org.apache.commons.io.function.IOSupplier;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class ByteArrayTransformer implements Transformer {
+public class ByteArrayTransformer extends Transformer<byte[]> {
     private static final Logger LOGGER = LoggerFactory.getLogger(ByteArrayTransformer.class);
 
+    private static final int MAX_BUFFER_SIZE = 4096;
+
     @Override
     public void configureValueConverter(final Map<String, String> config, final AbstractConfig sourceConfig) {
         // For byte array transformations, ByteArrayConverter is the converter which is the default config.
     }
 
     @Override
-    public Stream<Object> getRecords(final IOSupplier<InputStream> inputStreamIOSupplier, final String topic,
-            final int topicPartition, final AbstractConfig sourceConfig, final long skipRecords) {
-
-        // Create a Stream that processes each chunk lazily
-        return StreamSupport.stream(new Spliterators.AbstractSpliterator<>(Long.MAX_VALUE, Spliterator.ORDERED) {
-            final byte[] buffer = new byte[4096];
-            InputStream inputStream;
+    public StreamSpliterator<byte[]> createSpliterator(final IOSupplier<InputStream> inputStreamIOSupplier,
+            final String topic, final int topicPartition, final AbstractConfig sourceConfig) {
+        return new StreamSpliterator<byte[]>(LOGGER, inputStreamIOSupplier) {
+            @Override
+            protected InputStream inputOpened(final InputStream input) {
+                return input;
+            }
 
-            {
-                try {
-                    inputStream = inputStreamIOSupplier.get(); // Open the InputStream once
-                } catch (IOException e) {
-                    LOGGER.error("Error closing stream: {}", e.getMessage(), e);
-                }
+            @Override
+            protected void doClose() {
+                // nothing to do.
             }
 
             @Override
-            public boolean tryAdvance(final java.util.function.Consumer<? super Object> action) {
+            protected boolean doAdvance(final Consumer<? super byte[]> action) {
+                final byte[] buffer = new byte[MAX_BUFFER_SIZE];
```

Comment:
There is a method on s3 object. objectSummary.getSize()
May be we can put a limit on size of the file ex : 10mb or 100mb or something, and based on that if file size is beyond, then it would be splitted ?

Reply:
Agreed to fix this in a different pr